### PR TITLE
Added more descriptions on the exceptions.

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: proposed/http-client/http-client.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ use Psr\Http\Client\Exception;
 
 /**
  * Thrown when the request cannot be completed because of network issues.
+ * Example when the target host is not available.
```
Comment (dbu) [inline proposed/http-client/http-client.md]: i would move that to the end of the docblock, and say "For example when the target host name can not be resolved or the connection failed."

File: proposed/http-client/http-client.md [hunk 2]
```diff
@@ -126,7 +127,9 @@ use Psr\Http\Message\RequestInterface;
 use Psr\Http\Client\Exception;
 
 /**
- * Thrown when a response was received but the request itself failed.
+ * Thrown when a response was received but the request itself failed. This exception MAY
+ * be thrown on HTTP response codes 4xx and 5xx. This exception MUST NOT be thrown when
```
Comment (dbu) [inline proposed/http-client/http-client.md]: not sure what CS fig uses, but imho its best to have a single line with the general info, then a blank line and then more information.

**Comment by dbu** [comment]
fully agree with the content and think those are valuable additions

**Comment by Nyholm** [comment]
Thank you for the review

**Pull request merged**